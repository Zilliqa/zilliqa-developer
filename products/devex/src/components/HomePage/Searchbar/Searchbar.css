#searchbar-dropdown {
  height: 50px;
  border: 1px solid var(--color-border);
  background-color: var(--color-searchbar);
  color: var(--color-subtext);
  box-shadow: none !important;
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}

.searchbar-ig .dropdown-menu.show {
  min-width: fit-content;
}

.searchbar-ig {
  margin-bottom: 0.5rem;
}

.searchbar-ig input {
  height: 50px !important;
  margin-left: 1px;
  border-left: 0 !important;
  border-right: 0 !important;
}

/* Header Searchbar */
#header-searchbar-ig #searchbar-dropdown {
  height: 45px;
}

#header-searchbar-ig input {
  height: 45px !important;
}