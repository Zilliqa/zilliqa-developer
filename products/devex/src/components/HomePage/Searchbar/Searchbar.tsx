import React, { useState } from 'react'
import { Form, InputGroup, Button, Dropdown, DropdownButton } from 'react-bootstrap'
import { useHistory, useLocation } from 'react-router-dom'

import { isValidAddr } from 'src/utils/Utils'

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faSearch } from '@fortawesome/free-solid-svg-icons'

import './Searchbar.css'

interface IProps {
  isHeaderSearchbar: boolean,
  isISSearchbar: boolean,
}

const Searchbar: React.FC<IProps> = ({ isHeaderSearchbar, isISSearchbar }) => {

  const history = useHistory()
  const location = useLocation()
  const [input, setInput] = useState("")
  const [searchType, setSearchType] = useState('Txn/Addr')

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => (
    setInput(e.target.value)
  )

  const handleSubmit = (e: React.SyntheticEvent) => {
    e.preventDefault()
    const trimmedInput = input.trim()
    switch (searchType) {
      case 'Txn/Addr':
        if (isValidAddr(trimmedInput))
          history.push({
            pathname: `/address/${trimmedInput}`,
            search: location.search
          })
        else
          history.push({
            pathname: `/tx/${trimmedInput}`,
            search: location.search
          })
        break
      case 'Tx Block':
        history.push({
          pathname: `/txbk/${trimmedInput}`,
          search: location.search
        })
        break
      case 'DS Block':
        history.push({
          pathname: `/dsbk/${trimmedInput}`,
          search: location.search
        })
        break
    }
    setInput('')
  }

  return <>
    <Form onSubmit={handleSubmit}>
      <InputGroup className="searchbar-ig" id={isHeaderSearchbar ? "header-searchbar-ig" : "searchbar-ig"}>
        {isISSearchbar
          ? <InputGroup.Prepend>
            <DropdownButton id='searchbar-dropdown' title={searchType}>
              <Dropdown.Item onClick={() => setSearchType('Txn/Addr')}>Txn/Addr</Dropdown.Item>
              <Dropdown.Item onClick={() => setSearchType('Tx Block')}>Tx Block</Dropdown.Item>
            </DropdownButton>
          </InputGroup.Prepend>
          :
          <InputGroup.Prepend>
            <DropdownButton id='searchbar-dropdown' title={searchType}>
              <Dropdown.Item onClick={() => setSearchType('Txn/Addr')}>Txn/Addr</Dropdown.Item>
              <Dropdown.Item onClick={() => setSearchType('Tx Block')}>Tx Block</Dropdown.Item>
              <Dropdown.Item onClick={() => setSearchType('DS Block')}>DS Block</Dropdown.Item>
            </DropdownButton>
          </InputGroup.Prepend>
        }
        <Form.Control type="text" value={input} autoFocus={!isHeaderSearchbar}
          placeholder={
            searchType === 'Txn/Addr'
              ? 'Search for a transaction or an address'
              : searchType === 'Tx Block'
                ? 'Search by Tx Block height'
                : 'Search by DS Block height'}
          onChange={handleChange} />
        <InputGroup.Append>
          <Button type="submit">
            {isHeaderSearchbar ? <FontAwesomeIcon icon={faSearch} /> : <div>Search</div>}
          </Button>
        </InputGroup.Append>
      </InputGroup>
    </Form>
  </>
}

export default Searchbar
