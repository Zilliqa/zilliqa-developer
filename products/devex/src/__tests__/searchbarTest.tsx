import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Searchbar from 'src/components/HomePage/Searchbar/Searchbar'

jest.mock("react-router-dom", () => {
  const orig = jest.requireActual("react-router-dom")
  return {
    ...orig,
    useLocation: () => ({})
  }
})

describe('Isolated Server Header <Searchbar />', () => {

  const searchbar = shallow(<Searchbar isISSearchbar={true} isHeaderSearchbar={true} />)

  it('matches snapshot', () => {
    expect(toJson(searchbar)).toMatchSnapshot()
  })

})

describe('Non-Isolated Server Header <Searchbar />', () => {

  const searchbar = shallow(<Searchbar isISSearchbar={false} isHeaderSearchbar={true} />)

  it('matches snapshot', () => {
    expect(toJson(searchbar)).toMatchSnapshot()
  })

})
describe('Isolated Server Non-Header <Searchbar />', () => {

  const searchbar = shallow(<Searchbar isISSearchbar={true} isHeaderSearchbar={false} />)

  it('matches snapshot', () => {
    expect(toJson(searchbar)).toMatchSnapshot()
  })

})
describe('Non-Isolated Server Non-Header <Searchbar />', () => {

  const searchbar = shallow(<Searchbar isISSearchbar={false} isHeaderSearchbar={false} />)

  it('matches snapshot', () => {
    expect(toJson(searchbar)).toMatchSnapshot()
  })

})