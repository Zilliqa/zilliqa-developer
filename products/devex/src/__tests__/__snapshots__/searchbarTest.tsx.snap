// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Isolated Server Header <Searchbar /> matches snapshot 1`] = `
<Fragment>
  <Form
    inline={false}
    onSubmit={[Function]}
  >
    <InputGroup
      className="searchbar-ig"
      id="header-searchbar-ig"
    >
      <InputGroupPrepend>
        <DropdownButton
          id="searchbar-dropdown"
          title="Txn/Addr"
        >
          <DropdownItem
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "SafeAnchor",
                "render": [Function],
              }
            }
            disabled={false}
            onClick={[Function]}
          >
            Txn/Addr
          </DropdownItem>
          <DropdownItem
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "SafeAnchor",
                "render": [Function],
              }
            }
            disabled={false}
            onClick={[Function]}
          >
            Tx Block
          </DropdownItem>
        </DropdownButton>
      </InputGroupPrepend>
      <FormControl
        autoFocus={false}
        onChange={[Function]}
        placeholder="Search for a transaction or an address"
        type="text"
        value=""
      />
      <InputGroupAppend>
        <Button
          active={false}
          disabled={false}
          type="submit"
          variant="primary"
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  512,
                  512,
                  Array [],
                  "f002",
                  "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                ],
                "iconName": "search",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
        </Button>
      </InputGroupAppend>
    </InputGroup>
  </Form>
</Fragment>
`;

exports[`Isolated Server Non-Header <Searchbar /> matches snapshot 1`] = `
<Fragment>
  <Form
    inline={false}
    onSubmit={[Function]}
  >
    <InputGroup
      className="searchbar-ig"
      id="searchbar-ig"
    >
      <InputGroupPrepend>
        <DropdownButton
          id="searchbar-dropdown"
          title="Txn/Addr"
        >
          <DropdownItem
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "SafeAnchor",
                "render": [Function],
              }
            }
            disabled={false}
            onClick={[Function]}
          >
            Txn/Addr
          </DropdownItem>
          <DropdownItem
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "SafeAnchor",
                "render": [Function],
              }
            }
            disabled={false}
            onClick={[Function]}
          >
            Tx Block
          </DropdownItem>
        </DropdownButton>
      </InputGroupPrepend>
      <FormControl
        autoFocus={true}
        onChange={[Function]}
        placeholder="Search for a transaction or an address"
        type="text"
        value=""
      />
      <InputGroupAppend>
        <Button
          active={false}
          disabled={false}
          type="submit"
          variant="primary"
        >
          <div>
            Search
          </div>
        </Button>
      </InputGroupAppend>
    </InputGroup>
  </Form>
</Fragment>
`;

exports[`Non-Isolated Server Header <Searchbar /> matches snapshot 1`] = `
<Fragment>
  <Form
    inline={false}
    onSubmit={[Function]}
  >
    <InputGroup
      className="searchbar-ig"
      id="header-searchbar-ig"
    >
      <InputGroupPrepend>
        <DropdownButton
          id="searchbar-dropdown"
          title="Txn/Addr"
        >
          <DropdownItem
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "SafeAnchor",
                "render": [Function],
              }
            }
            disabled={false}
            onClick={[Function]}
          >
            Txn/Addr
          </DropdownItem>
          <DropdownItem
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "SafeAnchor",
                "render": [Function],
              }
            }
            disabled={false}
            onClick={[Function]}
          >
            Tx Block
          </DropdownItem>
          <DropdownItem
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "SafeAnchor",
                "render": [Function],
              }
            }
            disabled={false}
            onClick={[Function]}
          >
            DS Block
          </DropdownItem>
        </DropdownButton>
      </InputGroupPrepend>
      <FormControl
        autoFocus={false}
        onChange={[Function]}
        placeholder="Search for a transaction or an address"
        type="text"
        value=""
      />
      <InputGroupAppend>
        <Button
          active={false}
          disabled={false}
          type="submit"
          variant="primary"
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  512,
                  512,
                  Array [],
                  "f002",
                  "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                ],
                "iconName": "search",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
        </Button>
      </InputGroupAppend>
    </InputGroup>
  </Form>
</Fragment>
`;

exports[`Non-Isolated Server Non-Header <Searchbar /> matches snapshot 1`] = `
<Fragment>
  <Form
    inline={false}
    onSubmit={[Function]}
  >
    <InputGroup
      className="searchbar-ig"
      id="searchbar-ig"
    >
      <InputGroupPrepend>
        <DropdownButton
          id="searchbar-dropdown"
          title="Txn/Addr"
        >
          <DropdownItem
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "SafeAnchor",
                "render": [Function],
              }
            }
            disabled={false}
            onClick={[Function]}
          >
            Txn/Addr
          </DropdownItem>
          <DropdownItem
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "SafeAnchor",
                "render": [Function],
              }
            }
            disabled={false}
            onClick={[Function]}
          >
            Tx Block
          </DropdownItem>
          <DropdownItem
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "SafeAnchor",
                "render": [Function],
              }
            }
            disabled={false}
            onClick={[Function]}
          >
            DS Block
          </DropdownItem>
        </DropdownButton>
      </InputGroupPrepend>
      <FormControl
        autoFocus={true}
        onChange={[Function]}
        placeholder="Search for a transaction or an address"
        type="text"
        value=""
      />
      <InputGroupAppend>
        <Button
          active={false}
          disabled={false}
          type="submit"
          variant="primary"
        >
          <div>
            Search
          </div>
        </Button>
      </InputGroupAppend>
    </InputGroup>
  </Form>
</Fragment>
`;
